processor/elasticapmprocessor/processor.go

```diff
@@ -76,6 +76,23 @@ func (p *TraceProcessor) ConsumeTraces(ctx context.Context, td ptrace.Traces) er
 			}
 			// Traces signal never need to be routed to service-specific datasets
 			p.enricher.Config.Resource.DeploymentEnvironment.Enabled = false
+
+			// Iterate through spans to find errors in span events
+			scopeSpans := resourceSpan.ScopeSpans()
+			for j := 0; j < scopeSpans.Len(); j++ {
+				spans := scopeSpans.At(j).Spans()
+				for k := 0; k < spans.Len(); k++ {
+					span := spans.At(k)
+					events := span.Events()
+					for l := 0; l < events.Len(); l++ {
+						event := events.At(l)
+						if routing.IsErrorEvent(event.Attributes()) {
```

Comment:
Doubt: we are saying here that only spans in ECS mode are processed as error events.
Is this correct?

Reply:
Yes, should only be applied in `ecs` mode. In `otel native`, we comply with otel conventions and do not store error events in dedicated datastreams, see https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/otel-data/src/main/resources/index-templates.